Remove ENABLE_WEB_CRYPTO directive

annevk opened the pull request:
#### cab109e7b57c869ae4dd216c35c4fdeac4bcd3c2
<pre>
Remove ENABLE_WEB_CRYPTO directive
<a href="https://bugs.webkit.org/show_bug.cgi?id=266968">https://bugs.webkit.org/show_bug.cgi?id=266968</a>

Reviewed by Michael Catanzaro.

Also address a couple tiny nits, in particular enum class type
declarations and namespace closing comments.

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WebCore/PAL/pal/PlatformGTK.cmake:
* Source/WebCore/PAL/pal/PlatformWPE.cmake:
* Source/WebCore/bindings/js/SerializedScriptValue.cpp:
(WebCore::isTypeExposedToGlobalObject):
(WebCore::unwrapCryptoKey):
(WebCore::CloneSerializer::dumpIfTerminal):
(WebCore::CloneSerializer::write):
(WebCore::CloneDeserializer::readCryptoKey):
(WebCore::CloneDeserializer::readTerminal):
* Source/WebCore/crypto/CommonCryptoUtilities.cpp:
* Source/WebCore/crypto/CommonCryptoUtilities.h:
* Source/WebCore/crypto/CryptoAlgorithm.cpp:
* Source/WebCore/crypto/CryptoAlgorithm.h:
* Source/WebCore/crypto/CryptoAlgorithmIdentifier.h:
* Source/WebCore/crypto/CryptoAlgorithmParameters.h:
* Source/WebCore/crypto/CryptoAlgorithmParameters.idl:
* Source/WebCore/crypto/CryptoAlgorithmRegistry.cpp:
* Source/WebCore/crypto/CryptoAlgorithmRegistry.h:
* Source/WebCore/crypto/CryptoKey.cpp:
* Source/WebCore/crypto/CryptoKey.h:
* Source/WebCore/crypto/CryptoKey.idl:
* Source/WebCore/crypto/CryptoKeyFormat.h:
* Source/WebCore/crypto/CryptoKeyPair.h:
* Source/WebCore/crypto/CryptoKeyPair.idl:
* Source/WebCore/crypto/CryptoKeyType.h:
* Source/WebCore/crypto/CryptoKeyUsage.h:
* Source/WebCore/crypto/CryptoKeyUsage.idl:
* Source/WebCore/crypto/JsonWebKey.h:
* Source/WebCore/crypto/JsonWebKey.idl:
* Source/WebCore/crypto/RsaOtherPrimesInfo.h:
* Source/WebCore/crypto/RsaOtherPrimesInfo.idl:
* Source/WebCore/crypto/SerializedCryptoKeyWrap.h:
* Source/WebCore/crypto/SubtleCrypto.cpp:
* Source/WebCore/crypto/SubtleCrypto.h:
* Source/WebCore/crypto/SubtleCrypto.idl:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmAES_CBC.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmAES_CBC.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmAES_CFB.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmAES_CFB.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmAES_CTR.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmAES_CTR.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmAES_GCM.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmAES_GCM.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmAES_KW.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmAES_KW.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmECDH.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmECDH.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmECDSA.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmECDSA.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmEd25519.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmEd25519.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmHKDF.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmHKDF.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmHMAC.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmHMAC.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmPBKDF2.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmPBKDF2.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmRSAES_PKCS1_v1_5.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmRSAES_PKCS1_v1_5.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmRSASSA_PKCS1_v1_5.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmRSASSA_PKCS1_v1_5.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmRSA_OAEP.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmRSA_OAEP.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmRSA_PSS.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmRSA_PSS.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmSHA1.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmSHA1.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmSHA224.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmSHA224.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmSHA256.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmSHA256.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmSHA384.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmSHA384.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmSHA512.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmSHA512.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmX25519.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmX25519.h:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmAES_CBCGCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmAES_CFBGCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmAES_CTRGCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmAES_GCMGCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmAES_KWGCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmECDHGCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmECDSAGCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmEd25519GCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmHKDFGCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmHMACGCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmPBKDF2GCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmRSAES_PKCS1_v1_5GCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmRSASSA_PKCS1_v1_5GCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmRSA_OAEPGCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmRSA_PSSGCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmRegistryGCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmX25519GCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoKeyECGCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoKeyOKPGCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoKeyRSAGCrypt.cpp:
* Source/WebCore/crypto/gcrypt/SerializedCryptoKeyWrapGCrypt.cpp:
* Source/WebCore/crypto/keys/CryptoAesKeyAlgorithm.idl:
* Source/WebCore/crypto/keys/CryptoEcKeyAlgorithm.idl:
* Source/WebCore/crypto/keys/CryptoHmacKeyAlgorithm.idl:
* Source/WebCore/crypto/keys/CryptoKeyAES.cpp:
* Source/WebCore/crypto/keys/CryptoKeyAES.h:
* Source/WebCore/crypto/keys/CryptoKeyAlgorithm.idl:
* Source/WebCore/crypto/keys/CryptoKeyEC.cpp:
* Source/WebCore/crypto/keys/CryptoKeyEC.h:
* Source/WebCore/crypto/keys/CryptoKeyHMAC.cpp:
* Source/WebCore/crypto/keys/CryptoKeyHMAC.h:
* Source/WebCore/crypto/keys/CryptoKeyOKP.cpp:
* Source/WebCore/crypto/keys/CryptoKeyOKP.h:
* Source/WebCore/crypto/keys/CryptoKeyRSA.cpp:
* Source/WebCore/crypto/keys/CryptoKeyRSA.h:
* Source/WebCore/crypto/keys/CryptoKeyRSAComponents.cpp:
* Source/WebCore/crypto/keys/CryptoKeyRSAComponents.h:
* Source/WebCore/crypto/keys/CryptoKeyRaw.cpp:
* Source/WebCore/crypto/keys/CryptoKeyRaw.h:
* Source/WebCore/crypto/keys/CryptoRsaHashedKeyAlgorithm.idl:
* Source/WebCore/crypto/keys/CryptoRsaKeyAlgorithm.idl:
* Source/WebCore/crypto/mac/CommonCryptoDERUtilities.cpp:
* Source/WebCore/crypto/mac/CommonCryptoDERUtilities.h:
* Source/WebCore/crypto/mac/CryptoAlgorithmAES_CBCMac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmAES_CFBMac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmAES_CTRMac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmAES_GCMMac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmAES_KWMac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmECDHMac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmECDSAMac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmEd25519Cocoa.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmHKDFMac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmHMACMac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmPBKDF2Mac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmRSAES_PKCS1_v1_5Mac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmRSASSA_PKCS1_v1_5Mac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmRSA_OAEPMac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmRSA_PSSMac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmRegistryMac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmX25519Cocoa.cpp:
* Source/WebCore/crypto/mac/CryptoDigestAlgorithm.h:
* Source/WebCore/crypto/mac/CryptoKeyECMac.cpp:
* Source/WebCore/crypto/mac/CryptoKeyMac.cpp:
* Source/WebCore/crypto/mac/CryptoKeyOKPCocoa.cpp:
* Source/WebCore/crypto/mac/CryptoKeyRSAMac.cpp:
* Source/WebCore/crypto/mac/CryptoUtilitiesCocoa.cpp:
* Source/WebCore/crypto/mac/CryptoUtilitiesCocoa.h:
* Source/WebCore/crypto/mac/SerializedCryptoKeyWrapMac.mm:
* Source/WebCore/crypto/openssl/CryptoAlgorithmAES_CBCOpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoAlgorithmAES_CFBOpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoAlgorithmAES_CTROpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoAlgorithmAES_GCMOpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoAlgorithmAES_KWOpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoAlgorithmECDHOpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoAlgorithmECDSAOpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoAlgorithmHKDFOpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoAlgorithmHMACOpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoAlgorithmPBKDF2OpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoAlgorithmRSAES_PKCS1_v1_5OpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoAlgorithmRSASSA_PKCS1_v1_5OpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoAlgorithmRSA_OAEPOpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoAlgorithmRSA_PSSOpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoAlgorithmRegistryOpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoKeyECOpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoKeyRSAOpenSSL.cpp:
* Source/WebCore/crypto/openssl/OpenSSLUtilities.cpp:
* Source/WebCore/crypto/openssl/OpenSSLUtilities.h:
* Source/WebCore/crypto/openssl/SerializedCryptoKeyWrapOpenSSL.cpp:
* Source/WebCore/crypto/parameters/AesCbcCfbParams.idl:
* Source/WebCore/crypto/parameters/AesCtrParams.idl:
* Source/WebCore/crypto/parameters/AesGcmParams.idl:
* Source/WebCore/crypto/parameters/AesKeyParams.idl:
* Source/WebCore/crypto/parameters/CryptoAlgorithmAesCbcCfbParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmAesCtrParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmAesGcmParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmAesKeyParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmEcKeyParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmEcdhKeyDeriveParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmEcdsaParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmHkdfParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmHmacKeyParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmPbkdf2Params.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmRsaHashedImportParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmRsaHashedKeyGenParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmRsaKeyGenParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmRsaOaepParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmRsaPssParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmX25519Params.h:
* Source/WebCore/crypto/parameters/EcKeyParams.idl:
* Source/WebCore/crypto/parameters/EcdhKeyDeriveParams.idl:
* Source/WebCore/crypto/parameters/EcdsaParams.idl:
* Source/WebCore/crypto/parameters/HkdfParams.idl:
* Source/WebCore/crypto/parameters/HmacKeyParams.idl:
* Source/WebCore/crypto/parameters/Pbkdf2Params.idl:
* Source/WebCore/crypto/parameters/RsaHashedImportParams.idl:
* Source/WebCore/crypto/parameters/RsaHashedKeyGenParams.idl:
* Source/WebCore/crypto/parameters/RsaKeyGenParams.idl:
* Source/WebCore/crypto/parameters/RsaOaepParams.idl:
* Source/WebCore/crypto/parameters/RsaPssParams.idl:
* Source/WebCore/crypto/parameters/X25519Params.idl:
* Source/WebCore/dom/Document.cpp:
* Source/WebCore/dom/Document.h:
* Source/WebCore/dom/EmptyScriptExecutionContext.h:
* Source/WebCore/dom/ScriptExecutionContext.h:
* Source/WebCore/page/ChromeClient.h:
(WebCore::ChromeClient::unwrapCryptoKey const):
* Source/WebCore/page/Crypto.cpp:
(WebCore::Crypto::Crypto):
* Source/WebCore/page/Crypto.h:
* Source/WebCore/page/Crypto.idl:
* Source/WebCore/platform/GCrypt.cmake:
* Source/WebCore/platform/LocalizedStrings.cpp:
(WebCore::webCryptoMasterKeyKeychainLabel):
* Source/WebCore/platform/LocalizedStrings.h:
* Source/WebCore/platform/OpenSSL.cmake:
* Source/WebCore/workers/WorkerGlobalScope.cpp:
* Source/WebCore/workers/WorkerGlobalScope.h:
* Source/WebCore/worklets/WorkletGlobalScope.h:
* Source/WebKit/UIProcess/API/C/WKPage.cpp:
(WKPageSetPageNavigationClient):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::unwrapCryptoKey):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
* Source/WebKitLegacy/mac/WebCoreSupport/WebChromeClient.h:
* Source/WebKitLegacy/mac/WebCoreSupport/WebChromeClient.mm:
* Source/cmake/OptionsGTK.cmake:
* Source/cmake/OptionsPlayStation.cmake:
* Source/cmake/OptionsWPE.cmake:
* Source/cmake/OptionsWin.cmake:
* Source/cmake/WebKitFeatures.cmake:
* Tools/Scripts/webkitperl/FeatureList.pm:

Canonical link: <a href="https://commits.webkit.org/272561@main">https://commits.webkit.org/272561@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a4861571215c84efd2f326d28f6522c4f077feea

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/32105 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/10825 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/33906 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/34649 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/29099 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/32936 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/13182 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/8046 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/28675 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/32506 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/9134 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/28703 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/7927 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/8099 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/28618 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/35994 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/27591 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/29194 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/29075 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/34209 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/32192 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/8216 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/6167 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/32067 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/9857 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/38615 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7502 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/8862 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/8190 "Passed tests") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/8744 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/cmake/OptionsGTK.cmake:
```diff
@@ -303,14 +302,12 @@ if (ENABLE_DOCUMENTATION AND NOT GIDocgen_FOUND)
+endif ()
+if (LibGcrypt_VERSION VERSION_LESS 1.7.0)
+    message(FATAL_ERROR "libgcrypt 1.7.0 is required to enable Web Crypto API support.")
 endif ()
```
Since these deps are required now, please move the find_package() call for Libtasn1 to the top of the file in the alphabetized list of required packages, and just add the 1.7.0 version requirement to the libgcrypt find_package() call. We no longer need to have custom error messages and they don't make sense anymore since there's no way to disable the feature. Same comment applies to both OptionsGTK.cmake and OptionsWPE.cmake as well.

annevk committed:
Remove ENABLE_WEB_CRYPTO directive
https://bugs.webkit.org/show_bug.cgi?id=266968

Reviewed by Michael Catanzaro.

Also address a couple tiny nits, in particular enum class type
declarations and namespace closing comments.

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WebCore/PAL/pal/PlatformGTK.cmake:
* Source/WebCore/PAL/pal/PlatformWPE.cmake:
* Source/WebCore/bindings/js/SerializedScriptValue.cpp:
(WebCore::isTypeExposedToGlobalObject):
(WebCore::unwrapCryptoKey):
(WebCore::CloneSerializer::dumpIfTerminal):
(WebCore::CloneSerializer::write):
(WebCore::CloneDeserializer::readCryptoKey):
(WebCore::CloneDeserializer::readTerminal):
* Source/WebCore/crypto/CommonCryptoUtilities.cpp:
* Source/WebCore/crypto/CommonCryptoUtilities.h:
* Source/WebCore/crypto/CryptoAlgorithm.cpp:
* Source/WebCore/crypto/CryptoAlgorithm.h:
* Source/WebCore/crypto/CryptoAlgorithmIdentifier.h:
* Source/WebCore/crypto/CryptoAlgorithmParameters.h:
* Source/WebCore/crypto/CryptoAlgorithmParameters.idl:
* Source/WebCore/crypto/CryptoAlgorithmRegistry.cpp:
* Source/WebCore/crypto/CryptoAlgorithmRegistry.h:
* Source/WebCore/crypto/CryptoKey.cpp:
* Source/WebCore/crypto/CryptoKey.h:
* Source/WebCore/crypto/CryptoKey.idl:
* Source/WebCore/crypto/CryptoKeyFormat.h:
* Source/WebCore/crypto/CryptoKeyPair.h:
* Source/WebCore/crypto/CryptoKeyPair.idl:
* Source/WebCore/crypto/CryptoKeyType.h:
* Source/WebCore/crypto/CryptoKeyUsage.h:
* Source/WebCore/crypto/CryptoKeyUsage.idl:
* Source/WebCore/crypto/JsonWebKey.h:
* Source/WebCore/crypto/JsonWebKey.idl:
* Source/WebCore/crypto/RsaOtherPrimesInfo.h:
* Source/WebCore/crypto/RsaOtherPrimesInfo.idl:
* Source/WebCore/crypto/SerializedCryptoKeyWrap.h:
* Source/WebCore/crypto/SubtleCrypto.cpp:
* Source/WebCore/crypto/SubtleCrypto.h:
* Source/WebCore/crypto/SubtleCrypto.idl:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmAES_CBC.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmAES_CBC.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmAES_CFB.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmAES_CFB.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmAES_CTR.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmAES_CTR.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmAES_GCM.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmAES_GCM.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmAES_KW.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmAES_KW.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmECDH.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmECDH.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmECDSA.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmECDSA.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmEd25519.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmEd25519.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmHKDF.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmHKDF.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmHMAC.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmHMAC.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmPBKDF2.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmPBKDF2.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmRSAES_PKCS1_v1_5.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmRSAES_PKCS1_v1_5.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmRSASSA_PKCS1_v1_5.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmRSASSA_PKCS1_v1_5.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmRSA_OAEP.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmRSA_OAEP.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmRSA_PSS.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmRSA_PSS.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmSHA1.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmSHA1.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmSHA224.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmSHA224.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmSHA256.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmSHA256.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmSHA384.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmSHA384.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmSHA512.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmSHA512.h:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmX25519.cpp:
* Source/WebCore/crypto/algorithms/CryptoAlgorithmX25519.h:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmAES_CBCGCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmAES_CFBGCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmAES_CTRGCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmAES_GCMGCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmAES_KWGCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmECDHGCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmECDSAGCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmEd25519GCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmHKDFGCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmHMACGCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmPBKDF2GCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmRSAES_PKCS1_v1_5GCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmRSASSA_PKCS1_v1_5GCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmRSA_OAEPGCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmRSA_PSSGCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmRegistryGCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmX25519GCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoKeyECGCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoKeyOKPGCrypt.cpp:
* Source/WebCore/crypto/gcrypt/CryptoKeyRSAGCrypt.cpp:
* Source/WebCore/crypto/gcrypt/SerializedCryptoKeyWrapGCrypt.cpp:
* Source/WebCore/crypto/keys/CryptoAesKeyAlgorithm.idl:
* Source/WebCore/crypto/keys/CryptoEcKeyAlgorithm.idl:
* Source/WebCore/crypto/keys/CryptoHmacKeyAlgorithm.idl:
* Source/WebCore/crypto/keys/CryptoKeyAES.cpp:
* Source/WebCore/crypto/keys/CryptoKeyAES.h:
* Source/WebCore/crypto/keys/CryptoKeyAlgorithm.idl:
* Source/WebCore/crypto/keys/CryptoKeyEC.cpp:
* Source/WebCore/crypto/keys/CryptoKeyEC.h:
* Source/WebCore/crypto/keys/CryptoKeyHMAC.cpp:
* Source/WebCore/crypto/keys/CryptoKeyHMAC.h:
* Source/WebCore/crypto/keys/CryptoKeyOKP.cpp:
* Source/WebCore/crypto/keys/CryptoKeyOKP.h:
* Source/WebCore/crypto/keys/CryptoKeyRSA.cpp:
* Source/WebCore/crypto/keys/CryptoKeyRSA.h:
* Source/WebCore/crypto/keys/CryptoKeyRSAComponents.cpp:
* Source/WebCore/crypto/keys/CryptoKeyRSAComponents.h:
* Source/WebCore/crypto/keys/CryptoKeyRaw.cpp:
* Source/WebCore/crypto/keys/CryptoKeyRaw.h:
* Source/WebCore/crypto/keys/CryptoRsaHashedKeyAlgorithm.idl:
* Source/WebCore/crypto/keys/CryptoRsaKeyAlgorithm.idl:
* Source/WebCore/crypto/mac/CommonCryptoDERUtilities.cpp:
* Source/WebCore/crypto/mac/CommonCryptoDERUtilities.h:
* Source/WebCore/crypto/mac/CryptoAlgorithmAES_CBCMac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmAES_CFBMac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmAES_CTRMac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmAES_GCMMac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmAES_KWMac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmECDHMac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmECDSAMac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmEd25519Cocoa.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmHKDFMac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmHMACMac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmPBKDF2Mac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmRSAES_PKCS1_v1_5Mac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmRSASSA_PKCS1_v1_5Mac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmRSA_OAEPMac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmRSA_PSSMac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmRegistryMac.cpp:
* Source/WebCore/crypto/mac/CryptoAlgorithmX25519Cocoa.cpp:
* Source/WebCore/crypto/mac/CryptoDigestAlgorithm.h:
* Source/WebCore/crypto/mac/CryptoKeyECMac.cpp:
* Source/WebCore/crypto/mac/CryptoKeyMac.cpp:
* Source/WebCore/crypto/mac/CryptoKeyOKPCocoa.cpp:
* Source/WebCore/crypto/mac/CryptoKeyRSAMac.cpp:
* Source/WebCore/crypto/mac/CryptoUtilitiesCocoa.cpp:
* Source/WebCore/crypto/mac/CryptoUtilitiesCocoa.h:
* Source/WebCore/crypto/mac/SerializedCryptoKeyWrapMac.mm:
* Source/WebCore/crypto/openssl/CryptoAlgorithmAES_CBCOpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoAlgorithmAES_CFBOpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoAlgorithmAES_CTROpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoAlgorithmAES_GCMOpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoAlgorithmAES_KWOpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoAlgorithmECDHOpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoAlgorithmECDSAOpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoAlgorithmHKDFOpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoAlgorithmHMACOpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoAlgorithmPBKDF2OpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoAlgorithmRSAES_PKCS1_v1_5OpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoAlgorithmRSASSA_PKCS1_v1_5OpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoAlgorithmRSA_OAEPOpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoAlgorithmRSA_PSSOpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoAlgorithmRegistryOpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoKeyECOpenSSL.cpp:
* Source/WebCore/crypto/openssl/CryptoKeyRSAOpenSSL.cpp:
* Source/WebCore/crypto/openssl/OpenSSLUtilities.cpp:
* Source/WebCore/crypto/openssl/OpenSSLUtilities.h:
* Source/WebCore/crypto/openssl/SerializedCryptoKeyWrapOpenSSL.cpp:
* Source/WebCore/crypto/parameters/AesCbcCfbParams.idl:
* Source/WebCore/crypto/parameters/AesCtrParams.idl:
* Source/WebCore/crypto/parameters/AesGcmParams.idl:
* Source/WebCore/crypto/parameters/AesKeyParams.idl:
* Source/WebCore/crypto/parameters/CryptoAlgorithmAesCbcCfbParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmAesCtrParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmAesGcmParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmAesKeyParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmEcKeyParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmEcdhKeyDeriveParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmEcdsaParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmHkdfParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmHmacKeyParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmPbkdf2Params.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmRsaHashedImportParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmRsaHashedKeyGenParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmRsaKeyGenParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmRsaOaepParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmRsaPssParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmX25519Params.h:
* Source/WebCore/crypto/parameters/EcKeyParams.idl:
* Source/WebCore/crypto/parameters/EcdhKeyDeriveParams.idl:
* Source/WebCore/crypto/parameters/EcdsaParams.idl:
* Source/WebCore/crypto/parameters/HkdfParams.idl:
* Source/WebCore/crypto/parameters/HmacKeyParams.idl:
* Source/WebCore/crypto/parameters/Pbkdf2Params.idl:
* Source/WebCore/crypto/parameters/RsaHashedImportParams.idl:
* Source/WebCore/crypto/parameters/RsaHashedKeyGenParams.idl:
* Source/WebCore/crypto/parameters/RsaKeyGenParams.idl:
* Source/WebCore/crypto/parameters/RsaOaepParams.idl:
* Source/WebCore/crypto/parameters/RsaPssParams.idl:
* Source/WebCore/crypto/parameters/X25519Params.idl:
* Source/WebCore/dom/Document.cpp:
* Source/WebCore/dom/Document.h:
* Source/WebCore/dom/EmptyScriptExecutionContext.h:
* Source/WebCore/dom/ScriptExecutionContext.h:
* Source/WebCore/page/ChromeClient.h:
(WebCore::ChromeClient::unwrapCryptoKey const):
* Source/WebCore/page/Crypto.cpp:
(WebCore::Crypto::Crypto):
* Source/WebCore/page/Crypto.h:
* Source/WebCore/page/Crypto.idl:
* Source/WebCore/platform/GCrypt.cmake:
* Source/WebCore/platform/LocalizedStrings.cpp:
(WebCore::webCryptoMasterKeyKeychainLabel):
* Source/WebCore/platform/LocalizedStrings.h:
* Source/WebCore/platform/OpenSSL.cmake:
* Source/WebCore/workers/WorkerGlobalScope.cpp:
* Source/WebCore/workers/WorkerGlobalScope.h:
* Source/WebCore/worklets/WorkletGlobalScope.h:
* Source/WebKit/UIProcess/API/C/WKPage.cpp:
(WKPageSetPageNavigationClient):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::unwrapCryptoKey):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
* Source/WebKitLegacy/mac/WebCoreSupport/WebChromeClient.h:
* Source/WebKitLegacy/mac/WebCoreSupport/WebChromeClient.mm:
* Source/cmake/OptionsGTK.cmake:
* Source/cmake/OptionsPlayStation.cmake:
* Source/cmake/OptionsWPE.cmake:
* Source/cmake/OptionsWin.cmake:
* Source/cmake/WebKitFeatures.cmake:
* Tools/Scripts/webkitperl/FeatureList.pm:

Canonical link: https://commits.webkit.org/272561@main
